File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/memento/memento/Caretaker.java
package com.bigboxcode.designpattern.memento.memento;

import java.util.ArrayList;
import java.util.List;

public class Caretaker {

    private List<Memento> mementoList = new ArrayList<>();

    public void add(Memento memento) {
        mementoList.add(memento);
    }

    public Memento getByIndex(int index) {
        return mementoList.get(index);
    }

    public Memento getCurrent() {
        return mementoList.get(mementoList.size() - 1);
    }

    public void undo() {
        mementoList.remove(mementoList.size() - 1);
    }
}

File: design-patterns/ts/memento/general/caretaker.ts
// caretaker.ts

import Memento from "./memento";

class Caretaker {

    private mementoList: Memento[] = [];

    add(memento: Memento): void {
        this.mementoList.push(memento);
    }

    getByIndex(index: number): Memento {
        return this.mementoList[index];
    }

    getCurrent(): Memento {
        return this.mementoList[this.mementoList.length - 1];
    }

    undo(): void {
        this.mementoList.splice(-1, 1);
    }
}

export default Caretaker;